### ISSUE TITLE
[1.17.1] LootPool serialization doesn't save functions when no conditions are defined

### ISSUE THREAD

**Issue by P3pp3rF1y** [MinecraftForge/MinecraftForge#8182]
I have noticed that functions got removed from my json loot tables when rerunning `runData` in 1.17.1.

Tracked it down to serialization in LootPool

https://github.com/MinecraftForge/MinecraftForge/blob/1.17.x/patches/minecraft/net/minecraft/world/level/storage/loot/LootPool.java.patch#L120-L121

```
         if (!p_79094_.conditions.isEmpty()) {
            jsonobject.add("functions", p_79096_.serialize(p_79094_.functions));
         }
```

It incorrectly checks for whether there are conditions present before saving functions. In the patch it can be seen that vanilla properly checks the functions.
(I may PR the fix for this, but haven't done a forge PR in quite a while and don't have time to figure it out now so logging this to be sure it's fixed at some point)

**Comment by sciwhiz12** [comment]
Yep, a clear bug as it looks at the `f_79024_` (`conditions`) field instead of the `f_79026_` (`functions`) field before serializing the `functions` part of the loot pool JSON.

**Comment by ChampionAsh5357** [comment]
Closed by 8a38416e0fe78f169a24c544dfda4a4ba22aad37
